pkg/core/flag.go

```diff
@@ -188,3 +189,56 @@ func (u uuidFlag) String() string {
 }
 
 /// -- END UUID FLAG TYPE --
+
+// LabelResourceFlag /
+// Can only be a string for which there is a corresponding endpoint in cloudapi labels API
+// For example, https://api.ionos.com/cloudapi/v6/datacenters/{datacenterId}/servers/{serverId}/labels
+// As of Sep. 2022, the set of valid strings are "datacenter", "server", "volume", "ipblock", "snapshot".
+// NOTE: Track progress of https://github.com/spf13/pflag/issues/236 : Might be implemented in pflag
+type LabelResourceFlag struct {
+	Value   string
+	Allowed []string
+}
+
+func newLabelResourceFlag(defaultValue string) *LabelResourceFlag {
+	return &LabelResourceFlag{
+		Value: defaultValue,
+		Allowed: []string{
+			cloudapiv6.DatacenterResource,
+			cloudapiv6.ServerResource,
+			cloudapiv6.VolumeResource,
+			cloudapiv6.IpBlockResource,
+			cloudapiv6.SnapshotResource,
+		},
+	}
+}
+
+func (a *LabelResourceFlag) Set(p string) error {
```

Comment:
Maybe SetIfIncluded? Simple setters don't usually return errors

Reply:
While I agree this function name could be better, this is an override of the `Set` function that is called by `pflag`. I'm not sure if there's any "entry-points" into `pflag` where we can change its behavior to also check if the flag's values are what we expect them to be